[Zzazan] 1주차 zzazanstagram 기능 구현

wbluke opened the pull request:
[구현한 기능 목록]
- 프론트엔드 뷰
- 회원가입
- 로그인
- 게시글 저장
- 게시글 조회

패키지 구조를 어떻게 가져가야 할지 잘 몰라서 많이 고민했습니다. 패키지 구조에 대해서 조언 부탁드려요 :)
리뷰 잘 부탁드립니다-! 감사합니다~

wbluke committed:
init: project 생성

wbluke committed:
init: 프론트 뷰 파일 초기 세팅

JunHoPark93 committed:
[#18] feat: 공통 패키지 및 엔티티 정의

wbluke committed:
Merge pull request #19 from wbluke/feature/18-common-package-setting

[#18] feat: 공통 패키지 및 엔티티 정의

hyogenie-v committed:
[#15] feat: 게시글 모델 생성

1. Article 엔티티 클래스를 추가한다
2. Article 엔티티의 하위 VO 클래스를 추가한다
3. 그래들에 의존성을 추가한다

starkim06 committed:
[#7] feat: Member 엔티티 정의

wbluke committed:
[#20] refactor: index.html의 눈 모양 아이콘을 게시글 작성 아이콘으로 변경

wbluke committed:
[#20] feat: article을 작성하는 기본적인 페이지 작성

wbluke committed:
[#20] feat: article-edit.html의 자바스크립트 틀 작성

wbluke committed:
Merge pull request #21 from wbluke/feature/14-article-frontend

Feature/14 article frontend

starkim06 committed:
[#8] feat: 회원가입 기능

Closes #8

wbluke committed:
Merge pull request #23 from wbluke/feature/6-member-setting

Feature/6 member setting

hyogenie-v committed:
[#15] feat: 게시글 작성 기능 구현

1. 게시글을 작성하는 기능을 추가한다
   1-1. ArticleController 클래스를 추가한다
   1-2. POST 요청 시, 작성된 게시글을 저장한다
   1-3. ArticleRequest DTO 클래스를 추가한다
   1-4. ArticleRepository 클래스를 추가한다
2. 게시글 작성 기능에 대한 컨트롤러의 테스트 메서드를 추가한다

hyogenie-v committed:
Merge branch 'develop' into feature/14-article-create

wbluke committed:
Merge pull request #25 from wbluke/feature/14-article-create

[#15] 게시글 작성 기능 구현 PR

wbluke committed:
[#22] refactor: 사용하지 않는 js console.log 삭제

starkim06 committed:
[#26] refactor: 8월 13일 리뷰에 따른 리팩토링 적용

JunHoPark93 committed:
Merge pull request #27 from wbluke/refactor/26-review

[#26] refactor: 8월 13일 리뷰에 따른 리팩토링 적용

wbluke committed:
[#22] feat: zzazanstagram logo navBar에 추가 및 다른 페이지에도 적용

hyogenie-v committed:
[#29] refactor: 8월 13일 게시글 관련 리뷰에 따른 리팩토링 적용

hyogenie-v committed:
[#30] feat: 게시글 작성 페이지와 서버 시스템 연결

starkim06 committed:
[#10] feat: 로그인 기능

Closes #10

hyogenie-v committed:
[#15] feat: 게시글 조회 기능 구현

1. ArticleAssembler 클래스에 DTO로 변환하는 메서드를 추가한다
2. ArticleController에서 서비스 로직을 분리한다
3. 불필요한 어노테이션을 삭제한다

hyogenie-v committed:
[#31] feat: 게시글 조회 페이지와 서버 시스템 연결

starkim06 committed:
[#10] refactor: 테스트 중복 제거

wbluke committed:
[#22] feat: signup 페이지 뷰 작성

hyogenie-v committed:
Merge branch 'develop' into feature/14-article-show

wbluke committed:
Merge pull request #35 from wbluke/feature/14-article-show

[#15] 게시글 조회 기능 구현 PR

wbluke committed:
Merge branch 'develop' into feature/6-signup-frontend

wbluke committed:
Merge pull request #33 from wbluke/feature/6-signup-frontend

Feature/6 signup frontend

JunHoPark93 committed:
Merge branch 'develop' into feature/10-user-login

wbluke committed:
Merge pull request #37 from wbluke/feature/10-user-login

Feature/10 user login

JunHoPark93 committed:
[#39] refactor: 8/14 리팩토링

JunHoPark93 committed:
[#39] refactor: 8/14 리팩토링

JunHoPark93 committed:
Merge pull request #40 from wbluke/refactor/39-review

Refactor/39 review

wbluke committed:
[#41] 회원가입 form의 placeholder 한글로 변경 및 정규식 pattern 추가

wbluke committed:
[#41] 회원가입 페이지의 배치 정밀 수정

wbluke committed:
[#41] feat: validate error 메시지 tooltip으로 추가

wbluke committed:
Merge pull request #42 from wbluke/feature/6-signup-frontend-refactor

[#41] 회원가입 페이지 뷰 리팩토링

wbluke committed:
[#24] feat: login.html 페이지 뷰 작성

wbluke committed:
Merge pull request #43 from wbluke/feature/6-login-frontend

[#24] 로그인 페이지 뷰 작성

wbluke committed:
[#44] refactor: domain entity에 column 어노테이션 추가

wbluke committed:
[#44] refactor: 예외 관련 테스트 케이스 추가 및 관련 프로덕션 로직 추가

wbluke committed:
[#44] refactor: LoginController, LoginService를 Member관련 클래스에서 분리

wbluke committed:
[#44] refactor: Article entity에 Member entity 연관관계 매핑 추가

wbluke committed:
[#44] refactor: 프론트엔드 로직과 백엔드 로직 연동

wbluke committed:
[#44] refactor: Member package 구조 변경

wbluke committed:
Merge pull request #45 from wbluke/feature/6-review-refactor-#44

[#44] 8/15 리팩토링 및 프론트엔드와 백엔드 연결

wbluke committed:
[#44] refactor: MemberTest에 assertThat import

wbluke committed:
Merge pull request #46 from wbluke/feature/6-review-refactor-#44-fix

[#44] refactor: MemberTest에 assertThat import

wbluke committed:
Merge pull request #47 from wbluke/develop

[DEMO] ver.0.0.1 merge

wmakerjun commented on src/main/resources/templates/signup.html:
```diff
@@ -0,0 +1,188 @@
+        el.setCustomValidity('');
+    }
+    
+    email.addEventListener("input", function (event) {
```
event객체를 사용하지 않을꺼면 굳이 파라미터로 넣어주지 않아도 좋을 것 같아요 :)

wmakerjun commented on src/main/resources/templates/signup.html:
```diff
@@ -0,0 +1,188 @@
+<script src="https://stackpath.bootstrapcdn.com/bootstrap/4.1.3/js/bootstrap.bundle.min.js" integrity="sha384-pjaaA8dDz/5BgdFUPX6M/9SUZv4d12SUPF0axWc+VRZkx5xU3daN+lYb49+Ax+Tl" crossorigin="anonymous"></script>
+<script>
+    
+    const form = document.querySelector("#signup-from");
```
지금 회원가입 버튼 눌렀을 때 이벤트 리스너가 동작되지 않습니다. ```querySelector``` 안에 id tag name을 다시 확인해주세요 ㅠㅠ

wmakerjun commented on src/main/resources/templates/login.html:
```diff
@@ -0,0 +1,117 @@
+<script src="https://cdnjs.cloudflare.com/ajax/libs/pace/1.0.2/pace.min.js"></script>
+<script src="https://unpkg.com/popper.js/dist/umd/popper.min.js"></script>
+<script src="https://stackpath.bootstrapcdn.com/bootstrap/4.1.3/js/bootstrap.bundle.min.js" integrity="sha384-pjaaA8dDz/5BgdFUPX6M/9SUZv4d12SUPF0axWc+VRZkx5xU3daN+lYb49+Ax+Tl" crossorigin="anonymous"></script>
+<script>
```
사용하지 않는 스크립트 태그는 삭제해주세요 :)

wmakerjun commented on src/main/resources/templates/login.html:
```diff
@@ -0,0 +1,117 @@
+                            </div>
+                        </div>
+	                    
+                        <div class="card" style="width: 350px; display: inline-block; position: absolute; margin-top: 100px">
```
html 태그에 스타일을 직접 부여하기보다는 따로 css 파일을 만들어서 분리하면 더 좋을 것 같아요~!

wmakerjun commented on src/main/resources/templates/index.html:
```diff
@@ -0,0 +1,206 @@
+            document.querySelector(".advanced-search").classList.add("active")
+        } else {
+            document.querySelector(".advanced-search").classList.remove("active")
+            $('.advanced-search').removeClass("active");
```
이건 저의 실수인데...ㅠㅠ 200 line에 있는 코드와 중복된 제이쿼리 코드이기 때문에 201라인 코드는 삭제해주세요 😭

wmakerjun commented on src/main/resources/templates/article-edit.html:
```diff
@@ -0,0 +1,152 @@
+    function goBackToIndexPage(event) {
+        event.preventDefault();
+    
+        if (confirm("작성하신 게시물을 취소하고 돌아가시겠습니까?") == true) {
```
정확한 비교를 위해 == 보다는 ===을 이용하는게 좋습니다.
http://guswnsxodlf.github.io/javascript-equal-operator

wmakerjun commented on src/main/resources/templates/signup.html:
```diff
@@ -0,0 +1,188 @@
+    const passwordConfirm = document.querySelector("#signup-form-password-confirm");
+    const nickName = document.querySelector("#signup-form-nick-name");
+    const name = document.querySelector("#signup-form-name");
+    const profile = document.querySelector("#signup-form-profile");
```
사용하지 않는 변수는 지워주세요 :)

wmakerjun reviewed (changes requested):
전반적으로 HTML5에서 제공하는 기본적인 form validation과 api를 매우 잘 활용하고 있습니다. 한 메서드의 길이도 길지 않고 깔끔하게 잘 구현하고 있어요~!  💯 😄 
앞으로 기능이 추가되면서 자바스크립트 코드가 많아질텐데요~ 코드량이 늘어남에 따라 투두리스트 때 작성한 것처럼 구조화를 진행하면, 여러 크루들이 함께 협업하는데 더 큰 도움이 될 것 같아요.

wbluke committed:
[#50] fix: 사용하지 않는 코드 제거 및 의도하지 않은 오타 등의 코드로직 수정

wbluke committed:
[#50] refactor: html 태그에 각각 적용된 style 속성을 별도의 css 파일로 분리:zzazan.css

kang-hyungu commented on src/main/java/com/woowacourse/zzazanstagram/model/article/domain/Article.java:
```diff
@@ -0,0 +1,51 @@
+    @JoinColumn(name = "author", nullable = false, foreignKey = @ForeignKey(name = "fk_article_to_member"))
+    private Member author;
+
+    protected Article() {
```
상속을 허용하는게 아니라면 private 접근제어자를 추천드립니다.

hyogenie-v replied:
하이버네이트에서 지연로딩을 구현할 때 엔티티 클래스를 상속해서 프록시 객체를 만든다고 알고 있습니다!
그래서 기본 생성자에 private을 지정해두게 되면 프록시 클래스에서 상속을 받지 못하기 때문에 protected로 두었습니다~

kang-hyungu replied:
아 맞네요 그부분을 제가 놓쳤네요 👍

kang-hyungu commented on src/main/java/com/woowacourse/zzazanstagram/model/article/domain/Article.java:
```diff
@@ -0,0 +1,51 @@
+        this.author = author;
+    }
+
+    public static Article from(final Image image, final Contents contents, final Member author) {
```
지금은 생성자로 충분해 보이는데 팩토리 메서드를 따로 만든 이유가 있을까요?

kang-hyungu commented on src/main/java/com/woowacourse/zzazanstagram/model/article/domain/vo/Contents.java:
```diff
@@ -0,0 +1,28 @@
+import javax.persistence.Lob;
+
+@Embeddable
+public class Contents {
```
Contents를 클래스로 새로 정의할 필요가 있을까요?
Article 클래스에서 String contents 필드로 작성하면 코드가 더 간결해지지 않을까요?

hyogenie-v replied:
Contents를 일급컬렉션으로 빼놓은 이유는 **필드를 일관성 있게 구성하고 싶어서** 였습니다!
만약 어떤 필드는 원시타입으로 관리하고 있는데 다른 필드가 일급컬렉션으로 관리하고 있다면 데이터를 관리할 때 혼란스러울 것이라고 생각했습니다 :D

woowahanCU replied:
일급컬렉션이요? wrapper class를 말씀하신걸까요?

kang-hyungu replied:
데이터 관리할 때 어떤 점이 어려울지 잘 모르겠네요;
예시나 좀더 자세한 상황을 알려주실 수 있을까요?

kang-hyungu commented on src/main/java/com/woowacourse/zzazanstagram/model/article/domain/vo/Image.java:
```diff
@@ -0,0 +1,44 @@
+
+    private String validateUrl(final String url) {
+        try {
+            HttpURLConnection.setFollowRedirects(false);
```
이미지가 http가 아닌 ftp에 있다면?
다른 통신 방법을 이용해야된다면 http에 의존하는 이 코드가 문제가 되지 않을까요?

hyogenie-v replied:
이 부분은 저희가 차후에 S3를 사용해서 이미지 파일을 관리할 예정에 있기 때문에 ftp는 고려하지 않았습니다!

하지만 나중에 다른 기능을 구현하는 경우에는 이런 부분도 생각을 해봐야겠네요.
알려주셔서 감사합니다!

woowahanCU replied:
크롬이 이제 ftp 지원을 안한다고 합니다 ㅎㅎ; 그러니 저희도.. 😏

https://www.chromestatus.com/feature/6246151319715840

kang-hyungu replied:
ftp는 예시구요, 도메인 코드에서 유효성 체크를 위해 http 통신을 할 필요가 있냐는 의견이에요
표현 계층에서 이미지가 존재하는지 결정해도 충분하지 않나요?
지금 이 유효성 체크는 이미지를 호출하는 외부 서비스가 응답이 없으면 같이 영향을 받습니다.
차라리 프론트에서 이미지를 못 불러오더라도 다른 기능은 동작하도록 하고, 이 객체는 이미지 위치만 전달하는 역할만 해도 충분해보입니다.

kang-hyungu commented on src/main/java/com/woowacourse/zzazanstagram/model/article/dto/ArticleRequest.java:
```diff
@@ -0,0 +1,45 @@
+    private String image;
+
+    private String contents;
+    private String hashTag;
```
이 필드와 관련된 기능은 추가 예정인거죠? article 클래스에서 아직 관련 기능이 안보이네요 ㅎㅎ;

hyogenie-v replied:
넵 :-D
도메인 설계는 거시적으로 생각하긴 했는데 한 번에 적용할 수는 없을 것 같아서 선언만 해놓았어요~

kang-hyungu commented on src/main/java/com/woowacourse/zzazanstagram/model/article/service/ArticleService.java:
```diff
@@ -0,0 +1,45 @@
+    }
+
+    public List<ArticleResponse> getArticleResponses() {
+        List<Article> articles = articleRepository.findAllByOrderByIdDesc();
```
다음 단계 개발하면서 페이징을 추가해보세요!

kang-hyungu commented on src/main/java/com/woowacourse/zzazanstagram/model/common/BaseEntity.java:
```diff
@@ -0,0 +1,60 @@
+        return lastModifiedDate;
+    }
+
+    public void setId(Long id) {
```
setter 필요할까요?

hyogenie-v replied:
앗 변경했습니다!

kang-hyungu commented on src/main/java/com/woowacourse/zzazanstagram/model/member/domain/vo/Email.java:
```diff
@@ -0,0 +1,36 @@
+    private static final String EMAIL_REGEX = "^[_a-zA-Z0-9-.]+@[.a-zA-Z0-9-]+\\.[a-zA-Z]+$";
+
+    @Column(name = "email", unique = true)
+    private String email;
```
email 어노테이션을 적용하는건 어떨까요?

kang-hyungu commented on src/main/java/com/woowacourse/zzazanstagram/model/member/domain/vo/ProfileImage.java:
```diff
@@ -0,0 +1,48 @@
+
+    private String validateUrl(final String url) {
+        try {
+            HttpURLConnection.setFollowRedirects(false);
```
인프라스트럭처 계층 코드가 있네요

kang-hyungu commented on src/main/java/com/woowacourse/zzazanstagram/model/member/domain/vo/ProfileImage.java:
```diff
@@ -0,0 +1,48 @@
+        return url;
+    }
+
+    public ProfileImage updateUrl(String url) {
```
메서드명으로 update가 적절해보이진 않네요
valueOf 같이 불변 객체를 생성하는 의미가 담긴 메서드명을 찾아보는건 어떨까요

kang-hyungu commented on src/main/java/com/woowacourse/zzazanstagram/model/member/exception/MemberException.java:
```diff
@@ -0,0 +1,10 @@
+package com.woowacourse.zzazanstagram.model.member.exception;
+
+public class MemberException extends IllegalArgumentException {
```
Member와 관련된 모든 오류를 이 예외에서 다 처리하지 말고 좀더 세분화해보는건 어떨까요?
회원 로그인 시도할 때 비밀번호가 틀렸을 때 예외, 회원을 찾을 수 없을 때 발생하는 예외 등등으로 나누면 에러 메시지에 의존하지 않고 어떤 이슈인지 빠르게 파악 가능할거에요

kang-hyungu replied:
MemberException 세분화 잘 하셨어요 👍 
그런데 MemberException이 IllegalArgumentException를 상속하는게 맞을까요?

kang-hyungu commented on src/main/java/com/woowacourse/zzazanstagram/model/member/service/MemberService.java:
```diff
@@ -0,0 +1,52 @@
+        }
+    }
+
+    private void checkEnrolledNickName(String nickName) {
```
boolean existsByNickName(NickName nickName);
이 메서드를 memberRepository에 추가하면 코드가 좀더 간결해질거에요

kang-hyungu commented on src/test/java/com/woowacourse/zzazanstagram/model/article/domain/ArticleTest.java:
```diff
@@ -0,0 +1,58 @@
+    }
+
+    @ParameterizedTest
+    @ValueSource(strings = {"https://image.shutterstock.com/image-photo/white-transparent-leaf-on-mirror-600w-1029171697.jpg", "https://image.shutterstock.com/image-photo/bright-spring-view-cameo-island-600w-1048185397.jpg"})
```
```suggestion
    @ValueSource(strings = {
            "https://image.shutterstock.com/image-photo/white-transparent-leaf-on-mirror-600w-1029171697.jpg",
            "https://image.shutterstock.com/image-photo/bright-spring-view-cameo-island-600w-1048185397.jpg"
    })
```

woowahanCU replied:
IDE를 사용하시면 보이는 세로선은 이 margin까지만 코드를 작성하자는 컨벤션같은건데요.
옆으로 코드가 나열되면 가독성이 떨어집니다. 보통은 80 ~ 100이 기본입니다.

https://stackoverflow.com/questions/7756112/what-is-the-optimal-ide-text-editor-print-margin-when-programming

kang-hyungu reviewed (changes requested):
코드 깔끔하게 잘 구현하셨어요 👍 
패키지는 지금 구조로 가도 괜찮을 것 같네요
좀더 분리하자면 controller, (컨트롤러에서 사용하는)dto, member session 같이 표현 계층과 관련있는 패키지는 web으로 옮길 수도 있습니다.
몇 가지 코멘트 남겼으니 참고하시고 궁금한 점 있으면 코멘트 남기시거나 dm 주세요

wbluke committed:
Merge pull request #51 from wbluke/feature/14-review-refactor-frontend-#50

8/16 프론트엔드 리뷰 반영 리팩토링

JunHoPark93 committed:
[#52] refactor: 8/15 피드백 반영

JunHoPark93 committed:
[#52] refactor: 8/15 피드백 반영

dto에서 검사

woowahanCU commented on build.gradle:
```diff
@@ -0,0 +1,40 @@
+    mavenCentral()
+}
+
+dependencies {
```
gradlew 로 build 실행시에는 테스트를 skip하고 있어요. 
아래 설정 추가한 이후에 배포해보시겠어요?
```java
test {
    useJUnitPlatform()
}
...
dependencies {
...
    testImplementation 'org.junit.jupiter:junit-jupiter-api:5.3.1'
    testRuntimeOnly 'org.junit.jupiter:junit-jupiter-engine:5.3.1'
```

https://www.baeldung.com/junit-5-gradle

hyogenie-v committed:
[#55] refactor: 8/15 게시글 피드백 반영

woowahanCU commented on src/main/java/com/woowacourse/zzazanstagram/model/article/domain/Article.java:
```diff
@@ -0,0 +1,51 @@
+    }
+
+    public String image() {
+        return image.getUrl();
```
method name은 동작을 구체적으로 드러내는 동사를 사용하시는 것이 좋다고 생각합니다.

http://www.mimul.com/pebble/default/2013/05/04/1367639300999.html

woowahanCU commented on src/main/java/com/woowacourse/zzazanstagram/model/article/domain/vo/Image.java:
```diff
@@ -0,0 +1,44 @@
+package com.woowacourse.zzazanstagram.model.article.domain.vo;
```
model pacakge 하위에 controller, service, domain package가 있군요 🤔

wbluke committed:
Merge pull request #56 from wbluke/refactor/55-pr-review

[#55] refactor: 8/15 게시글 피드백 반영

woowahanCU commented on src/main/java/com/woowacourse/zzazanstagram/model/article/domain/vo/Image.java:
```diff
@@ -0,0 +1,44 @@
+        try {
+            HttpURLConnection.setFollowRedirects(false);
+            HttpURLConnection con = (HttpURLConnection) new URL(url).openConnection();
+            con.setRequestMethod("HEAD");
```
Image class에서 URL 리소스의 유효성을 검증하는 것이 좋을까요?
Image model이 HttpURLConnection class와 연관관계를 가지는 것보다 
Image 도메인 모델의 생애주기를 관리하는 repository의 구현체, 혹은 dao 등과 연관관계를 가지는 것은 어떨까요?

hyogenie-v committed:
Merge branch 'develop' into refactor/52-pr-review

hyogenie-v committed:
Merge pull request #53 from wbluke/refactor/52-pr-review

[#52] refactor: 8/15 피드백 반영

woowahanCU commented on src/main/java/com/woowacourse/zzazanstagram/util/SessionKeys.java:
```diff
@@ -0,0 +1,5 @@
+package com.woowacourse.zzazanstagram.util;
+
+public class SessionKeys {
+    public static final String MEMBER = "member";
```
이 클래스는 왜 Util package에 있는 것일까요?
그리고 Utility Class라면 private 생성자를 사용하길 effective java item4에서 가이드하고 있어요

https://medium.com/@biratkirat/learning-effective-java-item-4-4bc457fc5674

woowahanCU commented on src/main/java/com/woowacourse/zzazanstagram/model/member/controller/LoginController.java:
```diff
@@ -0,0 +1,34 @@
+        return "login";
+    }
+
+    @PostMapping("/login")
```
RequestMapping을 활용할 수 있겠네요.

woowahanCU commented on src/main/java/com/woowacourse/zzazanstagram/model/member/domain/vo/Name.java:
```diff
@@ -0,0 +1,40 @@
+        return name;
+    }
+
+    public Name updateName(String name) {
```
Name class이니, validateName, updatName 등의 Name도 중복이지 않을까요? 🙄

woowahanCU commented on src/main/java/com/woowacourse/zzazanstagram/model/member/domain/vo/ProfileImage.java:
```diff
@@ -0,0 +1,48 @@
+        try {
+            HttpURLConnection.setFollowRedirects(false);
+            HttpURLConnection con = (HttpURLConnection) new URL(url).openConnection();
+            con.setRequestMethod("HEAD");
```
Image class에 있는 로직과 중복이네요..? 🙄

woowahanCU commented on src/main/java/com/woowacourse/zzazanstagram/model/member/service/MemberService.java:
```diff
@@ -0,0 +1,52 @@
+
+    public void save(MemberSignUpRequest memberSignupRequest) {
+        checkEnrolledEmail(memberSignupRequest.getEmail());
+        checkEnrolledNickName(memberSignupRequest.getNickName());
```
매번 저장할때마다 조회쿼리가 여러번 발생하겠군요. 🙄

woowahanCU commented on src/test/java/com/woowacourse/zzazanstagram/model/RequestTemplate.java:
```diff
@@ -0,0 +1,66 @@
+@ActiveProfiles("test")
+@AutoConfigureWebTestClient
+@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT)
+public abstract class RequestTemplate {
```
👍

woowahanCU reviewed (approved):
테스트코드 깔끔하게 잘 작성하셨네요! 💯 
그리고 이슈도 잘 정리하고 있는 것으로 보이는데요. 
칸반보드를 활용하지 않다보니 현재 진행중인 이슈는 무엇인지 확인하기가 어렵네요 ㅠㅠ 
혹시 Projects 탭을 활용하지 않는 이유가 있을까요?

wbluke commented:
> 테스트코드 깔끔하게 잘 작성하셨네요! 💯
> 그리고 이슈도 잘 정리하고 있는 것으로 보이는데요.
> 칸반보드를 활용하지 않다보니 현재 진행중인 이슈는 무엇인지 확인하기가 어렵네요 ㅠㅠ
> 혹시 Projects 탭을 활용하지 않는 이유가 있을까요?

앗 내부적으로 ZenHub 사용해서 칸반보드 관리하고 있습니다! :) 기능이 많아서 좋은데 외부에선 보이지 않는다는 단점이 있네요ㅜㅜ

starkim06 committed:
[#57] refactor: 유틸 클래스(SessionKeys) 생성자 private 으로 선언

starkim06 committed:
[#57] feat: test 용 의존성 추가

기존에 gradlew 로 build 실행시에는 테스트를 skip하고 있어서
https://www.baeldung.com/junit-5-gradle 를 참고하여 의존성 추가

starkim06 committed:
[#57] refactor: Image 클래스에서 Http요청을 통해서 이미지 URL확인로직 제거

starkim06 committed:
[#57] refactor: RequestMapping을 활용하여 중복 Url제거

starkim06 committed:
[#57] refactor: validate 메서드 명 수정

starkim06 committed:
[#57] refactor: Email, NickName에 대해서 각각 요청하던 쿼리를 합침

기존의 existsByEmail, existsByNickName메서드를 통해서 각각 쿼리를 날리던
상황에서 existsByNickNameOrEmail를 사용하여 한 번만 쿼리를 날리도록 수정

starkim06 committed:
[#57] refactor: 메서드명 동사로 수정

Article, Member 클래스에서 Vo의 String 값을 요청하는 메서드명을 동사로 수정

wbluke commented:
> 전반적으로 HTML5에서 제공하는 기본적인 form validation과 api를 매우 잘 활용하고 있습니다. 한 메서드의 길이도 길지 않고 깔끔하게 잘 구현하고 있어요~! 💯 😄
> 앞으로 기능이 추가되면서 자바스크립트 코드가 많아질텐데요~ 코드량이 늘어남에 따라 투두리스트 때 작성한 것처럼 구조화를 진행하면, 여러 크루들이 함께 협업하는데 더 큰 도움이 될 것 같아요.

리뷰 주셔서 감사합니다! 주신 피드백은 모두 적용하였는데 구조화 작업은 아직 자바스크립트 실력이 부족하여 진행하지 못하였습니다ㅠㅠ 공부를 조금 더 하고나서 적용해 보겠습니다...!

wbluke committed:
[#57] refactor: SessionKeys를 web package로 이동

wbluke committed:
Merge pull request #58 from wbluke/refactor/57-review

[#57] refactor: 8/17 씨유 코치님 리뷰 반영

wbluke committed:
Merge pull request #59 from wbluke/develop

[DEMO] ver.0.0.2 merge (0.0.1 refactor)

kang-hyungu reviewed (approved):
피드백 반영 잘 하셨어요 👍 
몇 가지 추가 코멘트 남긴 건 참고하시고 다음 단계 진행하세요!

woowahanCU reviewed (commented):
https://github.com/woowacourse/miniprojects-2019/pull/4/commits/28e8bfb0ff8d501acec7e335b9a23c1b83645de9

어라, 이부분 추가안해줘도 테스트 동작하던가요? 🤔
```java
test {
    useJUnitPlatform()
}
```

JunHoPark93 commented:
> [28e8bfb](https://github.com/woowacourse/miniprojects-2019/commit/28e8bfb0ff8d501acec7e335b9a23c1b83645de9)
> 
> 어라, 이부분 추가안해줘도 테스트 동작하던가요? 🤔
> 
> ```java
> test {
>     useJUnitPlatform()
> }
> ```

앗 죄송합니다 빼먹었네요 ㅠ 다시 올려놓겠습니다

JunHoPark93 committed:
[#60] refactor: gradle test 추가

wbluke committed:
Merge pull request #61 from wbluke/refactor/60-gradle-test

[#60] refactor: gradle test 추가

wbluke committed:
Merge pull request #62 from wbluke/develop

[DEMO] hotfix v.0.0.2

dasistHYOJIN commented:
굿

hyogenie-v reviewed (approved):
아주 멋있고 아름다운 코드와 구조네요
승인합니다
